cstrml/staking/src/lib.rs

```diff
@@ -1545,39 +1631,71 @@ impl<T: Trait> Module<T> {
         }
     }
 
-    /// Reward a given block author by a specific amount. Add reward to the block author's
-    fn reward_author(stash: &T::AccountId, reward: BalanceOf<T>) -> PositiveImbalanceOf<T> {
-          Self::make_payout(stash, reward).unwrap_or(<PositiveImbalanceOf<T>>::zero())
-    }
-
-    /// Reward a given (maybe)validator by a specific amount. Add the reward to the validator's, and its
-    /// guarantors' balance, pro-rata based on their exposure, after having removed the validator's
-    /// pre-payout cut.
-    fn reward_validator(stash: &T::AccountId, reward: BalanceOf<T>) -> PositiveImbalanceOf<T> {
-        // let reward = reward.saturating_sub(off_the_table);
-        let mut imbalance = <PositiveImbalanceOf<T>>::zero();
-        let validator_cut = if reward.is_zero() {
-            Zero::zero()
-        } else {
-            let exposure = Self::stakers(stash);
-            let total = exposure.total.max(One::one());
-            let total_rewards = Self::validators(stash).guarantee_fee * reward;
-            let mut guarantee_rewards = Zero::zero();
-
-            for i in &exposure.others {
-                let per_u64 = Perbill::from_rational_approximation(i.value, total);
-                // Reward guarantors
-                guarantee_rewards += per_u64 * total_rewards;
-                imbalance.maybe_subsume(Self::make_payout(&i.who, per_u64 * total_rewards));
-            }
+    fn do_reward_stakers(
+        validator_stash: T::AccountId,
+        era: EraIndex,
+    ) -> DispatchResult {
+        // Validate input data
+        let current_era = CurrentEra::get().ok_or(Error::<T>::InvalidEraToReward)?;
+        ensure!(era <= current_era, Error::<T>::InvalidEraToReward);
```

Comment:
This actually won't happend? cause the `era == current_era`? But nevermind, this is double guard.

Reply:
This could happen since `era` is a value input by anyone by client api.

Reply:
got that